### PR TITLE
fix(kit): textMaskOptions pipe checked countryCode in previous value

### LINKED ISSUE #419: [BUG] `InputPhone` removing value leave country code
# 🐞 Bug report

### Description

When you hit backspace and remove whole number of `InputPhone`, then click away — country code remains in the input, although it is treated as empty value (label flies down). It disappears on the next focus. Can be reproduced on demo portal:
https://taiga-ui.dev/components/input-phone

### PR DESCRIPTION
**Pull request by imitrakov**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
-   [x] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [x] Bugfix
-   [ ] Feature
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Other... Please describe:

## What is the current behavior?

Closes #419 

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: projects/kit/components/input-phone/input-phone.component.ts [hunk 1]
```diff
@@ -96,14 +96,21 @@ export class TuiInputPhoneComponent
...
-        pipe: value => {
+        pipe: (value, config) => {
             if (this.allowText) {
                 return value;
             }
 
-            return value === '' && this.focused && !this.readOnly
-                ? `${this.countryCode} `
-                : value.replace(/-$/, '');
+            if (value === '' && this.focused && !this.readOnly) {
```
Comment (waterplea) [inline projects/kit/components/input-phone/input-phone.component.ts:104]: I think it's better to just reset native input to empty value upon leaving the control. You can set `nativeFocusableElement.value = ''` or create a setter for `get nativeValue` and do it in `onActiveZone` at the bottom of it.

**Review by waterplea — CHANGES_REQUESTED** [review]
Hi! Thank you for this PR. Check out my comment, I think I've spotted an easier way to fix it :)

File: projects/kit/components/input-phone/input-phone.component.ts [hunk 2]
```diff
@@ -156,6 +156,14 @@ export class TuiInputPhoneComponent
             : this.search || '';
     }
 
+    set nativeValue(value: string) {
+        setTimeout(() => {
```
Comment (waterplea) [inline projects/kit/components/input-phone/input-phone.component.ts:160]: @imitrakov  Why do you need the setTimeout here?

Comment (imitrakov) [reply projects/kit/components/input-phone/input-phone.component.ts:160]: Or return as I did before, when check  countryCode in pipe textMask? :)

**Comment by imitrakov** [comment]
Hi! Check out my comment, please :)

File: projects/kit/components/input-phone/input-phone.component.ts (hunk 2 above)
Comment (waterplea) [reply projects/kit/components/input-phone/input-phone.component.ts:160]: I see now. Previously active zone worked a little different and by the time pipe kicked in `focused` was already `false`. Looks like it's not as easy as I though 🤔

**Comment by waterplea** [comment]
Hi @imitrakov. Thank you for fixing this, but it turned out to be a tip of a bigger issue we have with `ActiveZone`. If you leave control with Tab key it would work fine but clicking away causing this issue — that's because by the time `ActiveZone` fires `false` from `mousedown` the focus is still in the zone. This is one of the trickier parts of the library so I'm going to have to close this PR and make my own fixing the root cause. Thank you for helping us uncover it!

**Pull request closed without merging**